### DISCUSSION TITLE
PRs Piled Up. Is this project on hold?

### DISCUSSION THREAD

**Discussion by douglasg14b** [CorentinTh/it-tools#1400] [General]
I ask because there are some tool improvements that I could aim PRs at, but there's `257` PRs at the time of writing with the last merge happening in October.

Is this project on hold, or in need of maintainers? Or frozen for a refactor?

If there's an issue or PR that's relevant to this, can it be pinned?

**Comment by sharevb** [comment]
Hi, if you arr interested in up to date version of it-tools, I made kind of a fork here : https://github.com/sharevb/it-tools (https://sharevb-it-tools.vercel.app/ and docker images https://github.com/sharevb/it-tools/pkgs/container/it-tools)

**Reply by ajpanton** [reply]
I tested it, and I know this might seem minor, but the multitude of spelling mistakes in many of the extra tools feels very rushed and not serious. I also see no way to report these to be fixed, at least not via a discussion or issues page.

**Reply by sharevb** [reply]
Hi @ajpanton, I just made this kind of temporary fork for people waiting about some new tools or enhancements for months.
 
Even on the original project, I don't think, there is a simple way to report spelling mistake apart github issues. 
The only way to report spelling mistake is on the original PR or issue in my repo.

**Reply by douglasg14b** [reply]
Without knowing the standards for how code/PRs are merged & managed, I'd be hesitant with running this for risk of malicious FE code being present i those standards are not present or seriously lacking.

**Reply by sharevb** [reply]
Hi @ajpanton, looked at many tools in my fork, found one missing tool description, but not so many other hurting things. Some of my PR have been checked by @steffenrapp ; could you please give some examples ? many thanks. Sounds like, the only solution to report things better is to use Vercel toolbar based on the fact that I would need to add "reviewer" in my vercel deployment

**Reply by sharevb** [reply]
Hi @douglasg14b, there are many facts, 1/ I made many PRs (13 merged with no changes, 149 awaiting and some reviewed by @steffenrapp and others) ; 2/ it-tools is only client side, no tracking except stats made by vercel ; 3/ CodeQL is run on the code (on original project also)
but I cannot prove some thing by just saying it

**Reply by steffenrapp** [reply]
Hi @sharevb it's great that you keep up the good work to support IT-Tools!
Unfortunately it seems that somehow all of the translations got lost in your fork.
I'm also a little bit concerned about the quality of the tools if there is no proper review process and if PRs seem to get merged without extensively testing them on desktop and mobile. I think it's crucial for a project like that to have top notch quality of a tool before releasing it as many people will rely on them to work as expected.
For example if you look at this screenshot the output looks kinda weird because it's too small. I have seen that in some of the tools and it just feels like not enough effort has been put into testing and fine tuning everything even if it works but the quality is not as high as it should be.
<img width="1251" alt="Bildschirmfoto 2024-12-12 um 17 32 13" src="https://github.com/user-attachments/assets/d40d0347-d04a-4a3b-b654-76a1040fb2d8" />

Also, I think you should question if every addition is really needed or if the functionality can be added to an existing tool (or does not make sense in this collection). There are so many tools now (which is absolutely awesome!) but it's kinda hard to navigate through the list and find the proper tool.
E.g. there is a tool "HEIC Converter" and "Image Formats Converter" which both serve a similar purpose but they're two tools ... I know there might be a reason for that but for an end user's point of view it doesn't make sense and should be one tool.
Same with "QR Code decoder" and "Barcode Reader" or the many unit converters.

Don't get me wrong, it's really good work but the usability can and should be improved.

Overall I still hope @CorentinTh will continue to push this project forward and add some maintainers (maybe you @sharevb ? 😃) who are willing to help, have some free resources and keep an eye on proper structure and quality of the tools.

**Reply by sharevb** [reply]
Hi @steffenrapp,

Thanks, and, I agree also but adding the fact that previews are no longer deployed, I decided to "quickly" merge all my PR (and some of other) because I use some of the tools and then allow to make report of bugs. (And you are a great reviewer)

Regarding number of tools, (for my use), I only use search box with the right keywords (could probably be improved too). But, right, some tools could be merged.
And yes, I removed other languages because I dont intent to make a total fork that will then be totaly unmergeable to original version. But I can add them back, it probably concern only tool name and description.

Regarding specific view bug, I dont reproduce on chrome or firefox. May be also a Ctrl+F5 sometime.

If you want to review some tools, feel free to report :)

Also, menu would need to refactor but this will probably be difficult to merge in original repo.

**Reply by sharevb** [reply]
And I must add that I found some bugs that will be corrected soon.

**Reply by steffenrapp** [reply]
Hi @sharevb oh yes of course, the missing previews make proper testing more difficult.

I'll take a closer look at some of the tools once I have some time (probably not too soon). I'm always glad to support great projects with the little I can provide.

**Reply by ajpanton** [reply]
> The only way to report spelling mistake is on the original PR or issue in my repo.

I can't see anything in your repo where one could report anything. Though I'm kind of new to this github thing, so maybe I'm missing something obvious.

Examples that I found so far:

### Multi link downloader:
I kind of understand what the description is trying to say, but it's worded in a weird way. Maybe:
"`The tool will asynchronously combine the contents of the provided links into a single zip file for you to download. (Requires an internet connection)`"
Also, this is nitpicking, but I would say
"`Add links separated by newlines...`"
"new lines" sounds like you need to have an empty line between each link, while "newline" is the name of the character you probably mean.

### AI Prompt Splitter:
- "priompts"
- "Splitted" is a non-standard form, should be "split". Because past and present are the same for this word, to avoid confusion in such a short phrase, I'd maybe use "divided" instead.
- "paramter"

### Math Formats Converter:
Doesn't work at all

### Week Numbers Converter:
**<ins>Description:</ins>** "`Compute Week Number in Year/Month vs Date`" This is very confusing, and also conveys no additional information to the title.
**<ins>Date to Week numbers:</ins>** This tool uses the American way of numbering weeks and the American way for arranging weekdays, both of which can change by country, neither of which is the ISO standard (though in this case at least America isn't alone in not following it).
**<ins>Year Week Number to Date:</ins>** This tool uses the ISO way of numbering weeks and the ISO way for arranging weekdays. Also, the naming could be better. For example "(Year and) Week number to date" (can skip year I think, see below).
**<ins>Month Week Number to Date:</ins>** This also uses the ISO standard. Naming could be better, for example "Week number in month to date" (since it's better to skip "year" here, for consistency, it can also be skipped in the previous).
**<ins>Also:</ins>** consistency with capitalization of the word "number" throughout this page.

### HDD calculator:
**<ins>Better description:</ins>** "`Convert claimed decimal HDD capacity to how software reports it`". There's no "real storage space" as it's all real.
**<ins>Alternative longer description:</ins>** "`Storage device manufacturers report capacities in decimal form (like GB), while operating systems and other software uses binary form (GiB), although incorrectly using the decimal unit (GB) for it. This tool converts decimal form capacity into various binary forms.`"
"Capacity in KB" lower case "k", and also all of these capacities should be kiB, MiB, GiB, TiB, PiB. Well, "KB" might actually be a non-standard way of saying kiB, but consistency.

### Timezone Converter:
The first "Timezones Date-Time Converter" tool was quite confusing. When choosing a date+time in the second dropdown dropdown, maybe specify "Choose a date/time interval".

I've only checked a tiny fraction of the tools available here, so there might be more.

And I'm only posting this in the hopes of making the tool better, as I'm also very thankful for it existing and someone actually putting in the work! It's just difficult to criticize something that someone has hade for free, but correcting mistakes will make it look and feel more serious and trustworthy.

**Reply by sharevb** [reply]
Hi @ajpanton, thanks for all these reportings I will integrate/check things soon, as a non native english speaker (and even in french, I am not a great "explainer"), I must admit that this will not come simply by myself.

**Reply by sharevb** [reply]
Hi @ajpanton, I modified and integrated some text in current version and make math converter work: https://sharevb-it-tools.vercel.app/
Thanks again for your reportings

**Comment by SamKr** [comment]
@sharevb: thanks for doing all that hard work on the pr's, and I'm really glad you released a fork as well! Cheers :)

If I come across spelling mistakes, I'll submit pr's on your repo for them.
